doc/AddingBlas3Op.md

```diff
@@ -0,0 +1,324 @@
+# Adding a new BLAS level 3 operation
+
+This document is meant to be a guide on how to add a new BLAS Level 3
+operation to SYCL-BLAS. It is mostly based on the work done when adding the
+TRSM algorithm to SYCL-BLAS.
+
+The steps described here are just a guideline and may vary when doing the
+actual implementation.
+
+## Add the operation interface
+
+The folder `include/interface` contains the public interface of SYCL-BLAS.
+The files in this folder contains the functions that users can call to 
+run the available blas operations.
+
+This folder has one file for each level of blas. For example, the `trsm`
+function will be located in `include/interface/blas3_interface.h`.
+
+When defining a new level 3 operation, the first step is to declare the
+user-facing function in this file, similar to the following:
```

Comment:
`internal::_trsm` is not quite user-facing, the `internal` namespace is not meant for users

Reply:
But you still need to declare the function in the `internal` namespace located `blas3_interface.h`, so in my view this is still visible to users in a way

Reply:
I think it needs to be obvious that there's also a `blas::_trsm` function - that would be the user facing one.

Reply:
Text improved in e3ba0b6a1aab686575ada22fe396d3384712cebf